amida-tech/indaba-client
# Bug/inba 782 and 804 jf router redirects

**DO NOT MERGE UNTIL CLEARED**. PR contains dependencies in alpha. This PR is for testing purposes until approved by Mike or react-router-redux gets a stable build. 

#### What does this PR do?
Properly offers redirects for various endpoints, basically redirecting if the user isn't logged in or isn't supposed to be at a particular page.

#### Related JIRA tickets:
INBA-852, INBA-782 and INBA-804.

#### How should this be manually tested?
After running the `yarn` install, you'll need to regression test every page with the following criteria:

- Users who **aren't** logged in can only access: `login` (with or without the realm), `reset-password` and `activate`. Attempts to go directly to any other page will redirect back to `login`.
- Users who are logged in but only as regular end users (roleID of 3), will be redirected to `task` if they attempt to access: `project` (including a direct project, ie `project/3`), `create-new-project`, `users`, and `subjects`.
- Admins should be able to access any page.

#### Background/Context

#### Screenshots (if appropriate):
